fabfile/utils.py

```diff
@@ -107,3 +124,32 @@ def add_fab_path_to_bashrc():
                 bashrc_text = re.sub(r'export FAB_PATH=[^\n]*', '', bashrc_text)
             bashrc.write(bashrc_text)
         local('echo "export FAB_PATH=$(pwd -P)" >> ~/.bashrc')
+
+
+def generate_printable_string(num_chars):
+    """
+    Generates a random string of printable characters
+    :param num_chars: number (int) of characters for the string to be
+    :return: String
+    """
+    result = ""
+    all_chars = string.printable.strip(string.whitespace)
+    while num_chars > 0:
+        result = "".join((result, choice(all_chars)))
+        num_chars -= 1
+    result = re.sub(r"'", "\\'", result)
+    return result
+
+
+def generate_ssh_keypair(pem_only=False):
+    """
+    Generates a 4096 bit ssh-keypair
+    :return: Tuple (str, str)
+    """
+    key = RSA.generate(4096)
+    public = key.publickey().exportKey('OpenSSH')
+    private = key.exportKey('PEM')
+    if pem_only:
+        return private
```

Comment:
it might be better practice to have this method always return the same output - right now it could return one variable or a tuple - maybe we would return a tuple with one object as `None` or we could make a namedtuple or even a simple class.